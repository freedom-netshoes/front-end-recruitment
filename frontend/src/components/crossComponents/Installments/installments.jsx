import React, { Component } from 'react'
import { PropTypes } from 'prop-types'
import CSSModules from 'react-css-modules'
import styles from './installments.less'
import Price from '../Price/price'

@CSSModules(styles)
export default class Installments extends Component {
  static defaultProps = {
    value: 0,
    installments: 0,
    type: 'simple'
  };

  static propTypes = {
    value: PropTypes.number,
    installments: PropTypes.number,
    type: PropTypes.oneOf(['simple', 'complete'])
  };

  render () {
    const {
      value,
      installments
    } = this.props
    const installmentValue = value / installments
    if (this.props.installments) {
      if (this.props.type === 'simple') {
        return (
          <div className='installments'>
            <span className='text'>ou</span>
            <span className='value'>
              {installments}
            </span>
            <span className='multiplier'>x</span>
            <Price {...{ value: installmentValue, type: 'full' }} />
          </div>
        )
      }
      return (
        <div className='installments'>
          <span className='text'>ou em at√©</span>
          <span className='value'>
            {installments}
          </span>
          <span className='multiplier'>x</span>
          <Price {...{ value: installmentValue, type: 'full' }} />
        </div>
      )
    }
  }
}
