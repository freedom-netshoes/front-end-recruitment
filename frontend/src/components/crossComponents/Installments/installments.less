@import url('../../../styles/_less-variables.less');
@import url('../../../styles/_less-mixins.less');
:global {
    #cart-shop {
        .subtotal-installments {
            .installments {
                color: @lighterGray;
                font-size: 12px;
                margin-top: 5px;
                .text {
                    margin: 0 5px;
                    text-transform: uppercase;
                }
                .multiplier {
                    margin: 0 2px;
                }
                .price {
                    .fontSemiBold();
                    font-size: 13px;
                }
            }
        }
    }
    #products {
        .installments {
            color: @lighterGray;
            font-size: 14px;
            .text {
                margin: 0 5px;
            }
            .multiplier {
                margin: 0 2px;
            }
            .price {
                  .fontSemiBold();
                font-size: 14px;
            }
        }
    }
}