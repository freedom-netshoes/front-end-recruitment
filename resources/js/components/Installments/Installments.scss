@import '../../../scss/shared';

:global(#products) .installments {
  color    : $light-gray;
  font-size: 14px;

  strong {
    @include open-sans-semi-bold();
  }
}

:global(.conclude) .installments {
  color         : $light-gray;
  font-size     : 12px;
  text-transform: uppercase;
}