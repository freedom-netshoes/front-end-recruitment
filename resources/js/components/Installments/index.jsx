import React, {Component, PropTypes} from 'react';
import CSSModules from 'react-css-modules';
import styles from './Installments.scss';

@CSSModules(styles)
class Installments extends Component {

  static defaultProps = {
    children: ''
  };

  static propTypes = {
    children: PropTypes.node
  };

  render () {
    return (
      <p styleName='installments'>
        {this.props.children}
      </p>
    );
  }
}

export default Installments;
