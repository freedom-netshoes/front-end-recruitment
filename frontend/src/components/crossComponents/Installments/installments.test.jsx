/* global describe, it, expect */
import React from 'react'
import Installments from './installments'
import { shallow } from 'enzyme'

describe('Testing <Installments /> component', () => {
  it('renders the Installments component', () => {
    const component = shallow(<Installments />)
    expect(component).toHaveLength(1)
  })

  it('Testing <Installments /> snapshop [Simple]', () => {
    const component = shallow(<Installments {...{
      value: 0,
      installments: 0,
      type: 'simple'
    }} />)
    expect(component).toMatchSnapshot()
  })

  it('Testing <Installments /> snapshop [Complete]', () => {
    const component = shallow(<Installments {...{
      value: 0,
      installments: 0,
      type: 'complete'
    }} />)
    expect(component).toMatchSnapshot()
  })
})
