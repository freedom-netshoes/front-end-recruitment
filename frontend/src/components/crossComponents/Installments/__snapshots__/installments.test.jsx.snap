// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Installments /> component Testing <Installments /> snapshop [Complete] 1`] = `null`;

exports[`Testing <Installments /> component Testing <Installments /> snapshop [Simple] 1`] = `null`;
